bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorBaseThingHandler.java

```diff
@@ -0,0 +1,282 @@
+        }
+    }
+
+    @Override
+    protected void processStateResponse(@Nullable SensorMessage stateResponse) {
+        if (stateResponse == null) {
+            return;
+        }
+        SensorConfig newSensorConfig = stateResponse.config;
+        sensorConfig = newSensorConfig != null ? newSensorConfig : new SensorConfig();
+        SensorState newSensorState = stateResponse.state;
+        sensorState = newSensorState != null ? newSensorState : new SensorState();
+
+        // Add some information about the sensor
+        if (!sensorConfig.reachable) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "Not reachable");
+            return;
+        }
+
+        if (!sensorConfig.on) {
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+
+        Map<String, String> editProperties = editProperties();
+        editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, stateResponse.swversion);
+        editProperties.put(Thing.PROPERTY_MODEL_ID, stateResponse.modelid);
+        editProperties.put(UNIQUE_ID, stateResponse.uniqueid);
+        ignoreConfigurationUpdate = true;
+        updateProperties(editProperties);
+
+        // Some sensors support optional channels
+        // (see https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Supported-Devices#sensors)
+        // any battery-powered sensor
+        if (sensorConfig.battery != null) {
+            createChannel(CHANNEL_BATTERY_LEVEL, ChannelKind.STATE);
+            createChannel(CHANNEL_BATTERY_LOW, ChannelKind.STATE);
+        }
+
+        createTypeSpecificChannels(sensorConfig, sensorState);
+
+        ignoreConfigurationUpdate = false;
+
+        // Initial data
+        updateChannels(sensorConfig);
+        updateChannels(sensorState, true);
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    protected void createChannel(String channelId, ChannelKind kind) {
+        ThingHandlerCallback callback = getCallback();
+        if (callback != null) {
+            ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+            ChannelTypeUID channelTypeUID;
+            switch (channelId) {
+                case CHANNEL_BATTERY_LEVEL:
+                    channelTypeUID = new ChannelTypeUID("system:battery-level");
+                    break;
+                case CHANNEL_BATTERY_LOW:
+                    channelTypeUID = new ChannelTypeUID("system:low-battery");
+                    break;
+                default:
+                    channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
+                    break;
+            }
+            Channel channel = callback.createChannelBuilder(channelUID, channelTypeUID).withKind(kind).build();
+            updateThing(editThing().withoutChannel(channelUID).withChannel(channel).build());
+        }
+    }
+
+    public void valueUpdated(ChannelUID channelUID, SensorConfig newConfig) {
```

Comment:
Is this really meant to be called from outside? I don't think so.`protected` would be better. Also JavaDoc is missing

Reply:
Changed and added JavaDoc...